sherpa/ui/tests/test_plot_pvalue.py

```diff
@@ -69,11 +64,326 @@ def test_plot_pvalue(make_data_path, clean_astro_ui, hide_log_output):
+    """Set the model and fit setup"""
+    ui.set_source(ui.gauss2d.g1 + ui.const2d.c1)
+
+    c1.c0 = 10
+    g1.fwhm = 4
+    g1.xpos = 4
+    g1.ypos = 3
+    g1.ampl = 100
+
+    ui.set_stat('cstat')
+    ui.set_method("simplex")
+
+    # Fit and check we are in the expected place (so this is a
+    # pre-condition for the tests).
+    #
+    ui.fit()
+    assert ui.calc_stat() == pytest.approx(362.9264257040166)
+
+
+def fake_psf_image(outfile):
+    """Create a PSF image
+
+    This is based on the output of image_psf() saved to a file
+    for the test_plot_pvalue_imgpsf_convolved case. As this is
+    a very-small file and it could change (to make the test more
+    meaningful) we do not add the data to the repository but
+    create it directly.
+
+    """
+
+    hdr = ['SIMPLE  =                    T /',
+           'BITPIX  =                  -64 / Bits per pixel',
+           'NAXIS   =                    2 /',
+           'NAXIS1  =                   10 /',
+           'NAXIS2  =                    8 /']
+
+    x1, x0 = np.mgrid[0:8, 0:10]
+    dx0 = (x0 - 5).flatten()
+    dx1 = (x1 - 4).flatten()
+
+    mdl = Gauss2D()
+    mdl.fwhm = 3
+    y = mdl(dx0, dx1)
+    y = y / y.max()
+
+    # Create the FITS structure
+    #
+    with open(outfile, "wb") as fh:
+        for h in hdr:
+            fh.write(f'{h:80s}'.encode('ascii'))
+
+        n = 2880 - (len(hdr) + 1) * 80
+
+        # assume we don't have enough lines to pass 1 block!
+        assert n > 0
+
+        fmt = f'{{:{n}s}}'
+        fh.write(fmt.format(' ').encode('ascii'))
+
+        fh.write('{:80s}'.format('END').encode('ascii'))
+
+        # FITS is big-endian
+        yb = y.astype('>f8').tobytes()
+        fh.write(yb)
+
+        n = 2880 - len(yb)
+        assert n > 0
+
+        fh.write(b'\0' * n)
+
+
+@pytest.fixture
+def setup_imgdata_file(tmp_path, recwarn):
+    """Use a model for the PSF"""
+
+    # Fake an image
+    #
+    x1, x0 = np.mgrid[0:8, 0:10]
+    ymod = 10 + 100 / ((x0 - 5.5)**2 + (x1 - 3.5)**2)
+
+    x0 = x0.flatten()
+    x1 = x1.flatten()
+    y = ymod.flatten()
+
+    y = y.astype(int)  # convert to integers
+
+    ui.load_arrays(1, x0, x1, y, (8, 10), ui.DataIMG)
+
+    # Sherpa can confuse astropy if you send in pathlib objects so
+    # convert to the raw filesystem path.
+    #
+    fname = tmp_path / "psf.fits"
+    infile = str(fname)
+    fake_psf_image(infile)
+
+    ui.load_psf('psf', infile)
+    psf.size = [10, 8]
+    psf.center = [5, 4]
+
+    # Just check where the warning comes from (pytest.warns didn't
+    # seem to remove the warning so it still triggered the general
+    # "check no warning" fixture, so use recwarn like this).
+    #
+    assert len(recwarn) == 0
+    ui.set_psf(psf)
+    assert len(recwarn) == 1
+
+    setup_imgdata_source()
+
+    assert len(recwarn) == 1
+    w = recwarn.pop()
+    assert issubclass(w.category, UserWarning)
+
+    # NOTE: there's a typo (missing space) in this warning
```

Comment:
I don't see the typo...

Reply:
There was but d30f60d0f5e81784d1ab8baaf4695b3212fb4c4c fixed it so this comment can go